src/main.js

```diff
@@ -55,23 +61,36 @@ if (crashReporterDirectory) {
 		}
 	}
 
-	// Crashes are stored in the temp directory by default, so we
+	// Crashes are stored in the crashDumps directory by default, so we
 	// need to change that directory to the provided one
-	console.log(`Found --crash-reporter-directory argument. Setting temp directory to be '${crashReporterDirectory}'`);
-	app.setPath('temp', crashReporterDirectory);
-
-	// Start crash reporter
-	const { crashReporter } = require('electron');
-	const productName = (product.crashReporter && product.crashReporter.productName) || product.nameShort;
-	const companyName = (product.crashReporter && product.crashReporter.companyName) || 'Microsoft';
-	crashReporter.start({
-		companyName: companyName,
-		productName: process.env['VSCODE_DEV'] ? `${productName} Dev` : productName,
-		submitURL: '',
-		uploadToServer: false
-	});
+	console.log(`Found --crash-reporter-directory argument. Setting crashDumps directory to be '${crashReporterDirectory}'`);
+	app.setPath('crashDumps', crashReporterDirectory);
+} else {
+	const appCenter = product.appCenter;
+	// Disable Appcenter crash reporting if
+	// * --crash-reporter-directory is specified
+	// * enable-crash-reporter runtime argument is set to 'false'
+	// * --disable-crash-reporter command line parameter is set
+	if (appCenter && argvConfig['enable-crash-reporter'] && !args['disable-crash-reporter']) {
+		const isWindows = (process.platform === 'win32');
+		const isLinux = (process.platform === 'linux');
+		const crashReporterId = argvConfig['crash-reporter-id'];
```

Comment:
@deepak1556 should we validate here that the `crashReporterId` is a `UUID`? After all, the user may have tampered with the value and we send this back to us.